src/model/model.ts

```diff
@@ -119,33 +124,84 @@ export abstract class AbstractModel<
       mergedParams.requestOpts.signal = params.requestOpts.signal;
     }
 
-    await Promise.allSettled(
-      this.events.onStart?.map((event) =>
-        Promise.resolve(
-          event({
-            timestamp: new Date().toISOString(),
-            modelType: this.modelType,
-            modelProvider: this.modelProvider,
-            params: mergedParams,
-            context: mergedContext,
-          })
-        )
-      ) ?? []
-    );
-
-    const cacheKey = await this.cacheKey(mergedParams);
-
-    try {
-      // Check the cache
-      const cachedResponse =
-        this.cache && (await Promise.resolve(this.cache.get(cacheKey)));
-      if (cachedResponse) {
-        const response: MResponse = {
-          ...cachedResponse,
-          cached: true,
-          cost: 0,
-          latency: Date.now() - start,
-        };
+    const spanName = getSpanName({
+      modelType: this.modelType,
+      ...mergedContext,
+    });
+
+    return Sentry.startSpan({ name: spanName, op: 'llm' }, async (span) => {
```

Comment:
I bet we could pas this in as an optional function in the constructor (`telemetryWrapper` or something) so that we don't need the sentry dependency in the library.  (Or pass it as an argument to `run`).

Reply:
yeah, i think that is the correct approach. however, we could use this asap, so i'm going to release it as a major version bump with a warning and we will make it optional in a future major release.

Reply:
@brianbegy @cfortuner i released it as v3 but as a prerelase with the `next` tag so we can use it, but it won't be used for anyone that does `npm install @dexaai/dexter`.